bench/Main.purs

```diff
@@ -161,38 +160,109 @@ htmlTableWrap caption benchmark = do
-  htmlTableWrap "runParser json smallJson" $ benchWith 1000
-    $ \_ -> runParser smallJson BenchParsing.json
-  htmlTableWrap "runTrampoline runParser json smallJson" $ benchWith 1000
-    $ \_ -> runTrampoline $ runParserT smallJson BenchParsing.json
-  htmlTableWrap "StringParser.runParser json smallJson" $ benchWith 500
-    $ \_ -> StringParser.runParser BenchStringParser.json smallJson
+
+  -- These inputs are too small for good measurement, but larger ones blow stack
+  -- log "<th><h2>digit 1000</h2></th>"
+  -- htmlTableWrap "runParser many digit 1000" $ benchWith 200
+  --   $ \_ -> runParser string23_1000 parse23
+  -- htmlTableWrap "StringParser many CodePoints.anyDigit 1000" $ benchWith 20
+  --   $ \_ -> StringParser.runParser parse23Points string23_1000
+  -- htmlTableWrap "StringParser many CodeUnits.anyDigit 1000" $ benchWith 200
+  --   $ \_ -> StringParser.runParser parse23Units string23_1000
+  -- htmlTableWrap "runParser manyRec digit 1000" $ benchWith 200
+  --   $ \_ -> runParser string23_1000 parse23Rec
+  -- htmlTableWrap "StringParser manyRec CodePoints.anyDigit 1000" $ benchWith 20
+  --   $ \_ -> StringParser.runParser parse23PointsRec string23_1000
+  -- htmlTableWrap "StringParser manyRec CodeUnits.anyDigit 1000" $ benchWith 200
+  --   $ \_ -> StringParser.runParser parse23UnitsRec string23_1000
+  -- htmlTableWrap "Regex.match \\d* 1000" $ benchWith 200
+  --   $ \_ -> Regex.match pattern23 string23_1000
+
+  log "<th><h2>digit 10000</h2></th>"
+  htmlTableWrap "runParser many digit 10000" $ benchWith 50
+    $ \_ -> runParser string23_10000 parse23
+  htmlTableWrap "runParser manyRec digit 10000" $ benchWith 50
+    $ \_ -> runParser string23_10000 parse23Rec
+  htmlTableWrap "runParser Array.many digit 10000" $ benchWith 50
+    $ \_ -> runParser string23_10000 (Array.many digit)
+  htmlTableWrap "StringParser manyRec CodePoints.anyDigit 10000" $ benchWith 20
+    $ \_ -> StringParser.runParser parse23PointsRec string23_10000
+  htmlTableWrap "StringParser manyRec CodeUnits.anyDigit 10000" $ benchWith 200
+    $ \_ -> StringParser.runParser parse23UnitsRec string23_10000
+  htmlTableWrap "Regex.match \\d* 10000" $ benchWith 200
+    $ \_ -> Regex.match pattern23 string23_10000
+
+  log "<th><h2>string 100000</h2></th>"
+  htmlTableWrap "runParser many string" $ benchWith 200
+    $ \_ -> runParser stringSkidoo_100000 parseSkidoo
+  htmlTableWrap "runParser manyRec string" $ benchWith 200
+    $ \_ -> runParser stringSkidoo_100000 parseSkidooRec
+  htmlTableWrap "Regex.match literal*" $ benchWith 200
+    $ \_ -> Regex.match patternSkidoo stringSkidoo_100000
+
+  log "<th><h2>sepBy 1000</h2></th>"
+  htmlTableWrap "runParser sepBy 1000" $ benchWith 50
+    $ \_ -> runParser string23_1000 $ sepBy anyChar (char '3')
+  htmlTableWrap "runParser sepByRec 1000" $ benchWith 50
+    $ \_ -> runParser string23_1000 $ sepByRec anyChar (char '3')
+
+  log "<th><h2>sepBy 10000</h2></th>"
+  -- sepBy not stack-safe
```

Comment:
A stack unsafe parser should be a bug, right?

Reply:
Yes I think so?

Reply:
I'm going to merge these benchmarks, then let's talk about it.